cms/djangoapps/contentstore/views/course.py

```diff
@@ -334,6 +334,27 @@ def course_search_index_handler(request, course_key_string):
         }), content_type=content_type, status=200)
 
 
+@require_GET
+def edevate_reindex_course(request, course_key_string):
+    """
+    The restful handler for course indexing.
+    GET
+        json: return status of indexing task
+    """
+    course_key = CourseKey.from_string(course_key_string)
+    content_type = "application/json; charset=utf-8"
+    with modulestore().bulk_operations(course_key):
+        try:
+            CoursewareSearchIndexer.do_course_reindex(modulestore(), course_key)
+        except SearchIndexingError as search_err:
+            return HttpResponse(dump_js_escaped_json({
+                "user_message": search_err.error_list
+            }), content_type=content_type, status=500)
```

Comment:
Shouldn't this be some kind of 4XX HTTP status? Returning 500 will result in not analysing real server errors properly.